src/execution/exec/execution_context.cpp

```diff
@@ -41,6 +42,21 @@ void ExecutionContext::RegisterThreadWithMetricsManager() {
 }
 
 void ExecutionContext::CheckTrackersStopped() {
```

Comment:
`CheckTrackersStopped` sounds like it's just returning a true/false rather than mutating state. Can we call this something like `EnsureTrackersStopped`vof `GuaranteeTrackersStopped` since that's what it's actually doing?

Reply:
Changed all instances of `CheckTrackersStopped` -> `EnsureTrackersStopped`.